src/CMakeLists.txt

```diff
@@ -1,3 +1,34 @@
+# Set version number from git-describe
+execute_process(COMMAND git describe --tags
+    OUTPUT_VARIABLE VERSION_STRING
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+string(REGEX REPLACE "(\n|\r)" "" VERSION_STRING "${VERSION_STRING}")
+string(REGEX REPLACE "-" "." VERSION_STRING "${VERSION_STRING}")
+string(REGEX REPLACE ".g.*" "" VERSION_STRING "${VERSION_STRING}")
+
+set(tomopy_VERSION "${VERSION_STRING}" CACHE STRING "Version of tomopy project" FORCE)
+mark_as_advanced(tomopy_VERSION)
+message(STATUS "tomopy version ${tomopy_VERSION}")
+
+foreach(_TYPE MAJOR MINOR PATCH)
+    set(tomopy_VERSION_${_TYPE} ${PROJECT_VERSION_${_TYPE}}
+        CACHE STRING "tomopy ${_TYPE} version" FORCE)
+    mark_as_advanced(tomopy_VERSION_${_TYPE})
+endforeach(_TYPE MAJOR MINOR PATCH)
+
+project(tomopy LANGUAGES C VERSION ${tomopy_VERSION})
+
+################################################################################
+
+include(MacroUtilities)
+include(Options)
+include(Compilers)
+include(GNUInstallDirs)
+include(ProjectSettings)
+include(BuildSettings)
+include(Packages)
+include(ClangFormat)
+
```

Comment:
I moved the `project` statement and the CMake module includes here: from `/` to `/src`. I did this because the directory of the `project` statement sets `${PROJECT_SOURCE_DIR}`. I thought it would be better to move the project statement than update all lines that contain `${PROJECT_SOURCE_DIR}`. The include statements are also moved because I observed that at least one of them needs to be declared after the project is declared otherwise certain variables are not set.

Reply:
No you should keep the project statement in the top CMakeLists.txt

Reply:
What if I replace `${PROJECT_SOURCE_DIR}` with `${CMAKE_CURRENT_LIST_DIR}` inside `/src/CMakeLists.txt`?